pkg/reconciler/taskrun/taskrun.go

```diff
@@ -359,6 +358,20 @@ func (c *Reconciler) prepare(ctx context.Context, tr *v1.TaskRun) (*v1.TaskSpec,
 		}
 	}
 
+	steps, err := resources.GetStepActionsData(ctx, *taskSpec, tr, c.PipelineClientSet)
+	switch {
+	case err != nil:
+		logger.Errorf("Failed to determine StepAction to use for taskrun %s: %v", tr.Name, err)
+		if resources.IsGetTaskErrTransient(err) {
```

Comment:
Are we reusing this function? Maybe we can rename it to something generic to apply to both StepActions and Tasks?
```suggestion
		if resources.IsGetTaskErrTransient(err) {
```

Reply:
Yeah, that makes sense,